common/message_algos/src/grouping/message_batcher.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{grouping::*, Envelope, Message, Timestamp};
+
+/// A concrete implementation of a `MessageGrouper`.
+pub struct MessageBatcher<T: Message> {
+    grouping_policy: Box<dyn GroupingPolicy<Message = T>>,
+    retirement_policy: Box<dyn RetirementPolicy<Message = T>>,
+    groups: Vec<MessageGroup<T>>,
+}
```

Comment:
I'm okay with the `MessageGrouper` trait but not with this generic implementation.

* It could work, but I feel that this is too complex for what we want to do in practice. The timeout for a batch can be computed at the creation of the batch (`system-time-timeout = first_message.event_time + event-time-batching-window + accepted_delay`). And the eviction criteria is only based on this timeout.